examples/pytorch/rgcn/link_utils.py

```diff
@@ -0,0 +1,280 @@
+"""
+Utility functions for link prediction
+Most code is adapted from authors' implementation of RGCN link prediction:
+https://github.com/MichSchli/RelationPrediction
+
+"""
+import numpy as np
+import torch as th
+import dgl
+
+# Utility function for building training and testing graphs
+
+def comp_deg_norm(g):
+    in_deg = g.in_degrees(range(g.num_nodes())).float()
+    norm = 1.0 / in_deg
+    norm[th.isinf(norm)] = 0
+    # convert to edge norm
+    g.ndata['norm'] = norm.unsqueeze(-1).long()
+    g.apply_edges(lambda edges : {'norm' : edges.dst['norm']})
+
+def get_subset_g(g, mask, num_rels, bidirected=False):
+    src, dst = g.edges()
+    sub_src = src[mask]
+    sub_dst = dst[mask]
+    sub_rel = g.edata['etype'][mask]
+
+    if bidirected:
+        sub_src, sub_dst = th.cat([sub_src, sub_dst]), th.cat([sub_dst, sub_src])
+        sub_rel = th.cat([sub_rel, sub_rel + num_rels])
+
+    sub_g = dgl.graph((sub_src, sub_dst), num_nodes=g.num_nodes())
+    sub_g.edata[dgl.ETYPE] = sub_rel
+
+    return sub_g
+
+def preprocess(g, num_rels):
```

Comment:
@VoVAllen Probably this could be the starting point for the `split_edges` utility.